code/controllers/subsystem/mapping/levels.dm

```diff
@@ -195,3 +226,34 @@
 		. = "[discriminator][num2hex(rand(1, 16 ** 4 - 1))]"
 	while(. in random_fluff_level_hashes)
 	random_fluff_level_hashes += .
+
+/**
+ * hooks us to SSmapping initializations; this should be called during New() for atoms.
+ *
+ * if no maploading can be hooked, we init immediately
+ * if Initialize() is in SSatoms, this crashes for safety as that should not happen.
+ */
+/datum/proc/hook_map_initializations()
```

Comment:
Why is this so low level? This is very niche, and only one type seems to use it.